### ISSUE TITLE
Support for volume notifications sent by gnome-settings-daemon (Ubuntu)

### ISSUE THREAD

**Issue by blueyed** [dunst-project/dunst#159]
It would be nice, if dunst would display notifications for volume control from gome-settings-daemon.

On Ubuntu, when using notify-osd instead of dunst, the following notifications get send:

```
method call sender=:1.4 -> dest=:1.67 serial=632 path=/org/freedesktop/Notifications; interface=org.freedesktop.Notifications; member=GetCapabilities
method return sender=:1.67 -> dest=:1.4 reply_serial=632
   array [
      string "body"
      string "body-markup"
      string "icon-static"
      string "image/svg+xml"
      string "x-canonical-private-synchronous"
      string "x-canonical-append"
      string "x-canonical-private-icon-only"
      string "x-canonical-truncation"
      string "private-synchronous"
      string "append"
      string "private-icon-only"
      string "truncation"
   ]
method call sender=:1.4 -> dest=:1.67 serial=633 path=/org/freedesktop/Notifications; interface=org.freedesktop.Notifications; member=Notify
   string "gnome-settings-daemon"
   uint32 0
   string "notification-audio-volume-high"
   string " "
   string ""
   array [
   ]
   array [
      dict entry(
         string "x-canonical-private-synchronous"
         variant             string "volume"
      )
      dict entry(
         string "value"
         variant             int32 88
      )
   ]
   int32 -1
method return sender=:1.67 -> dest=:1.4 reply_serial=633
   uint32 0
signal sender=:1.67 -> dest=:1.4 serial=384 path=/org/freedesktop/Notifications; interface=org.freedesktop.Notifications; member=NotificationClosed
   uint32 0
   uint32 1
```

But only after the query for capabilities has been replied to (which dunst appears to ignore).

It appears that this might be specific to Ubuntu (`x-canonical-private-synchronous`), and that it might work with vanilla gnome-settings-daemon?

**Comment by bebehei** [comment]
I researched this today.

> But only after the query for capabilities has been replied to (which dunst appears to ignore).

Dunst does not ignore the `GetCapabilities` method, `dbus-monitor` does not print the replies 😉 It turns out, that `x-canonical-private-synchronous` is the important capability, `private-synchronous` has no effect here.

> It appears that this might be specific to Ubuntu (`x-canonical-private-synchronous`), and that it might work with vanilla gnome-settings-daemon?

The full details are at [the NotifyOSD page](https://wiki.ubuntu.com/NotifyOSD).

Unity distinguishes between two types of notifications, (normal notifications and confirmations). A change in the volume is a confirmation and uses the `x-canonical-private-synchronous` hint.

NotifyOSD is Unity's own notification standard, but only available on Unity. Unity makes dumb assumptions about the notification daemon. At least, it queries for the capability, but it should go on and

1. query the capabilities
2. send a notification
    - as confirmation if one of `(x-canonical-)private-synchronous` cap is given
    - else as normal notification

Albeit `(x-canonical-)private-synchronous` is nowhere standardized, it would be a good addition in dunst, but there are other problems:

- confirmation notifications/bubbles are nowhere else a standard on DEs
- Instead of only checking for `x-canonical-private-synchronous`, checking for `private-synchronous` would make it not just canonical-specific
- Unity is not anymore the main DE in Ubuntu 18.04

There is no sense in adding support for `x-canonical-private-icon-only` nowadays and chances will decrease.

**Comment by tsipinakis** [comment]
#552 Attempts to implement this hint. Is anyone (still) using dunst with the default ubuntu shell such as this hint would be useful?

As I mentioned in [my comment](https://github.com/dunst-project/dunst/pull/552#issuecomment-431917332) I tried setting dunst up in ubuntu but `gnome-shell` grabs the notification address and couldn't figure out how to disable it.

**Comment by blueyed** [comment]
@tsipinakis 
I am not using Ubuntu myself anymore.

**Comment by bebehei** [comment]
Fixed in #552
